File: exemple.com/cgi-bin/hello.py
#!/usr/bin/env python3
import os
import cgi

print("Content-Type: text/html")
print()
print("<html><head><title>CGI Test</title></head><body>")
print("<h1>Hello from CGI!</h1>")

# Afficher les variables d'environnement
print("<h2>Environment Variables</h2>")
print("<pre>")
for key, value in os.environ.items():
    print(f"{key}: {value}")
print("</pre>")

# Afficher les paramètres CGI
form = cgi.FieldStorage()
print("<h2>CGI Parameters</h2>")
print("<pre>")
for key in form.keys():
    print(f"{key}: {form[key].value}")
print("</pre>")

print("</body></html>")

File: exemple.com/cgi-bin/hello.cpp
#include <iostream>

int main() {
    // Print the HTTP headers
    std::cout << "Content-Type: text/html\n\n";
    
    // Generate HTML content
    std::cout << "<html>\n";
    std::cout << "<head>\n";
    std::cout << "<title>CGI Script Example</title>\n";
    std::cout << "</head>\n";
    std::cout << "<body>\n";
    std::cout << "<h1>Hello, World!</h1>\n";
    std::cout << "<p>This is a simple CGI script written in C++.</p>\n";
    std::cout << "</body>\n";
    std::cout << "</html>\n";

    return 0;
}
